From 825761c2fe66e98d4d39f8adc8a19c4e5ec576a6 Mon Sep 17 00:00:00 2001
From: Yufei Gu <yufei.gu@cloudera.com>
Date: Thu, 10 Nov 2016 09:24:35 -0800
Subject: [PATCH 2014/2848] YARN-5862. TestDiskFailures.testLocalDirsFailures
 failed (Yufei Gu via Varun Saxena)

(cherry picked from commit c202a10923a46a6e7f7f518e6e3dbb6545dbb971)
(cherry picked from commit 06c8ebdb83ecae3bda7269e0bc9d133c1092e8da)

Change-Id: I411be0263d6cc56aa3e8ed8ed76863f24385b653
---
 .../hadoop/yarn/server/TestDiskFailures.java       |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/TestDiskFailures.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/TestDiskFailures.java
index c7e34d8..bf82ec5 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/TestDiskFailures.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/TestDiskFailures.java
@@ -244,7 +244,9 @@ private void verifyDisksHealth(boolean localORLogDirs, String expectedDirs,
     for (int i = 0; i < 10; i++) {
       Iterator<RMNode> iter = yarnCluster.getResourceManager().getRMContext()
                               .getRMNodes().values().iterator();
-      if ((iter.next().getState() != NodeState.UNHEALTHY) == isHealthy) {
+      // RMNode # might be zero because of timing related issue.
+      if (iter.hasNext() &&
+          (iter.next().getState() != NodeState.UNHEALTHY) == isHealthy) {
         break;
       }
       // wait for the node health info to go to RM
-- 
1.7.9.5

